src/lua/packagepath.lua

```diff
@@ -0,0 +1,50 @@
+-- packagepath.lua
+--
```

Comment:
I made this on purpose: I want a thing that says "this is an application, please treat its load paths in a special way". Notice that it not only adds .rocks to load path, but also the module directory as well, which enables you to always load subpackages using relative "paths" even when your `cwd()` is physically somewhere else.

Your suggestion to patch the loader only covers half of the thing I aim for (the .rocks). If you say that we should also patch the loader to magically handle the project root, then it likely goes against common intuition how regular load paths work (and I'd prefer to not have such magic)

As for what you write about "We cannot set parent directory as base for loaders" I didn't understand. Please elaborate. I only talk about application root here.

The "cases" shouldn't be different. We should push one preferred way for writing Tarantool apps, with "relocatable app" being the default (last time we discussed it with Kostya Osipov, he agreed with that). If the user doesn't want the "default" behavior, they are free to not use `package.setpath()` at all and be on their own. 

Another argument for not patching the loader is that it will modify its current behavior, and break something for current users. Which I specifically don't want to do after we've passed v2 beta as there is just too little time left to collect feedback and act on it. Instead my plan is to add this `package.setpath()` and vote for total removal of "recursive upwards rocks loader" in 3.0.